File: Android-Kotlin/driver/src/main/java/com/chatanoga/cab/driver/networking/http/GetRegisterInfo.kt
package com.chatanoga.cab.driver.networking.http

import com.chatanoga.cab.common.models.Driver
import com.chatanoga.cab.common.models.Service
import com.chatanoga.cab.common.networking.http.interfaces.HTTPRequest
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

class GetRegisterInfo(jwtToken: String): HTTPRequest() {
    override val path: String = "driver/get"
    init {
        this.params = mapOf("token" to jwtToken)
    }
}
@JsonClass(generateAdapter = true)
data class RegistrationInfo(
    val driver: Driver,
    val services: List<Service>
)

File: Ios-Swift/driver/driver/Networking/HTTP/GetRegisterInfo.swift
//
//  GetRegisterInfo.swift
//  driver
//


import Foundation


class GetRegisterInfo: HTTPRequest {
    var path: String = "driver/get"
    typealias ResponseType = RegistrationInfo
    var params: [String : Any]?
    
    init(jwtToken: String) {
        self.params = [
            "token": jwtToken
        ]
    }
}

struct RegistrationInfo: Codable {
    var driver: Driver
    var services: [Service] = []
}
